src/servers/src/mysql/server.rs

```diff
@@ -67,15 +72,28 @@ impl MysqlServer {
         })
     }
 
-    pub fn handle(
+    async fn handle(
         stream: TcpStream,
         io_runtime: Arc<Runtime>,
         query_handler: SqlQueryHandlerRef,
     ) -> Result<()> {
         info!("MySQL connection coming from: {}", stream.peer_addr()?);
         let shim = MysqlInstanceShim::create(query_handler, stream.peer_addr()?.to_string());
-        // TODO(LFC): Relate "handler" with MySQL session; also deal with panics there.
-        let _handler = io_runtime.spawn(AsyncMysqlIntermediary::run_on(shim, stream));
+
+        let (r, w) = stream.into_split();
+        let w = BufWriter::with_capacity(DEFAULT_RESULT_SET_WRITE_BUFFER_SIZE, w);
+        // TODO(LFC): Use `output_stream` to write large MySQL ResultSet to client.
+        let spawn_result = io_runtime
+            .spawn(AsyncMysqlIntermediary::run_on(shim, r, w))
+            .await;
+        match spawn_result {
+            Ok(run_result) => {
+                if let Err(e) = run_result {
+                    error!(e; "Internal error occurred during query exec, server actively close the channel to let client try next time.")
+                }
+            }
+            Err(e) => error!("IO runtime cannot execute task, error: {}", e),
```

Comment:
In these two error cases, why we don't return an error result?

Reply:
Seems to write these 2 errors to client requires further investigation upon writing raw mysql protocol bytes into client channel. Allow me to add a todo here.